packages/theme/src/react/with-theme.js

```diff
@@ -12,17 +12,29 @@ const getDisplayName = Component => {
   return Component.displayName || Component.name || 'Component'
 }
 
-export default function withTheme(BaseComponent) {
+export default function withTheme(BaseComponent, options = {}) {
+  const { forwardRef = true } = options
   const name = getDisplayName(BaseComponent)
 
-  const EnhancedComponent = (props, context = {}) => {
+  const EnhancedComponent = ({ forwardedRef, ...props }, context) => {
     const themeName = context.themeName || themeDefaultName
-    return <BaseComponent {...props} themeName={themeName} />
+    return <BaseComponent {...props} ref={forwardedRef} themeName={themeName} />
   }
 
-  EnhancedComponent.BaseComponent = BaseComponent
   EnhancedComponent.contextTypes = { themeName: PropTypes.string }
-  EnhancedComponent.displayName = `withTheme(${name})`
+  EnhancedComponent.propTypes = { forwardedRef: PropTypes.any }
 
+  if (forwardRef) {
+    const Forwarded = React.forwardRef(function forwardThemeRef(props, ref) {
```

Comment:
/me Thinking it's an interesting React API...

Reply:
the main issue i hit is, unlike a functional component, i don't have access to context in the forwardRef so i had to get tricky